pkg/oc/generate/cmd/newapp.go

```diff
@@ -859,6 +869,23 @@ func (c *AppConfig) Run() (*AppResult, error) {
 		if err != nil {
 			return nil, fmt.Errorf("can't setup %q: %v", p.From, err)
 		}
+		for _, obj := range accepted {
+			if bc, ok := obj.(*buildapi.BuildConfig); ok {
+				buildEnv := buildutil.GetBuildConfigEnv(bc)
+				buildEnv = app.JoinEnvironment(buildEnv, buildenv.List())
+				buildutil.SetBuildConfigEnv(bc, buildEnv)
```

Comment:
i'm not convinced this is the right place to be changing. 

--build-env already works today for non-pipeline buildconfigs and the codepath you're touching here would impact all buildconfigs i believe.

Something is unique about what new-app does for the "build strategy pipeline" vs what it does for "build strategy s2i/docker/etc" that is causing it to drop the --build-env env vars.

(at a minimum if this logic is added here, we need to find and remove the existing logic that is adding the env vars to all the other buildconfigs)

Reply:
thanks for the feedback, I will take a look at other strategies as well

Reply:
Yeah, to @bparees 's point take a look at https://github.com/openshift/origin/blob/master/pkg/oc/generate/app/app.go#L198-L233

Notice that the env's are set for source/docker build strategy, but not pipeline

Also, you may notice custom strategy is missing there but that is OK ... new-app doesn't deal with that strategy if memory serves me correctly